### PR TITLE
Add deletion docs

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: source/includes/dev/_deletion.md [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+
+```json
+{
+  "data": {
+    "type": "deletions",
+    "id": "62bd8d49cae089972ce81039",
+    "attributes": {
+      "userId": "62bd8d49cae089972ce81038",
+      "requestorUserId": "62bd8d49cae089972ce81038",
+      "status": "pending",
```
Comment (solomon-negusse) [inline source/includes/dev/_deletion.md:190]: `status` seems to be ignored in the route handler and is not present in the `Joi` delete creation config as well (in similar Python validator that would return  422 response in some conditions).

File: source/includes/dev/_deletion.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+```json
+{
+  "data": {
+    "type": "deletions",
+    "id": "62bd8d49cae089972ce81039",
+    "attributes": {
+      "userId": "62bd8d49cae089972ce81038",
+      "requestorUserId": "62bd8d49cae089972ce81038",
+      "status": "done",
```
Comment (solomon-negusse) [inline source/includes/dev/_deletion.md:267]: I know this is just an example but shouldn't the delete status of all the individual resources be set to `true` to be able to set overall deletion status to `done`.

**Review by solomon-negusse — CHANGES_REQUESTED** [review]

File: source/includes/dev/_deletion.md (hunk 1 above)
Comment (tiagojsag) [reply source/includes/dev/_deletion.md:190]: the router handler and Joi both validate the request input. This is an example of the serialized content returned in the response.

File: source/includes/dev/_deletion.md (hunk 2 above)
Comment (tiagojsag) [reply source/includes/dev/_deletion.md:267]: The goal here is to add a basic CRUD. There's definitely potential to add logic to all of this (for example, status could become a simple product of validating if all the boolean fields are true). However, that's outside of scope for now.

File: source/includes/dev/_deletion.md (hunk 1 above)
Comment (solomon-negusse) [reply source/includes/dev/_deletion.md:190]: Makes sense.  I thought that was request body which I now see the request doesn't require.

**Pull request merged**